lib/at_cmd/at_cmd.c

```diff
@@ -111,23 +134,114 @@ static int get_return_code(char *buf, struct return_state_object *ret)
 
 static void callback_worker(struct k_work *item)
 {
-	struct callback_work_item *data =
-		CONTAINER_OF(item, struct callback_work_item, work);
+	ARG_UNUSED(item);
+	struct resp_item resp;
 
-	if (data != NULL) {
-		data->callback(data->data);
+	while (k_msgq_get(&responses, &resp, K_NO_WAIT) == 0) {
+		if (resp.callback != NULL ||
+		    resp.buf != NULL) {
+			resp.callback(resp.buf);
+		}
+		if (resp.flags & AT_RESP_ALLOC) {
+			k_free(resp.buf);
+		}
 	}
+}
+
+
 
-	k_mem_slab_free(&rsp_work_items, (void **)&data);
+/* Distributes an AT response to the appropriate queue */
+static void distribute_resp(const struct cmd_item *cmd, struct resp_item *resp)
```

Comment:
Nit: I would call this function `dispatch_resp` instead.

Reply:
The function no longer exists